File: PolishPizza/frontend/src/service/menuItemService.ts
import { api } from '../api/api';
import { MenuItemResponse } from "../types/menuItem";

export const MenuItemService = {
    getAvailableMenuItems: async (): Promise<MenuItemResponse[]> => {
        try {
            const response = await api.get<MenuItemResponse[]>('/menu/menuItems/available');
            
            return response.data;
        } catch (error) {
            console.error('Menu-Service Error:', error);
            return [];
        }
    },

    getMenuItemById: async (id: String): Promise<MenuItemResponse | null> => {
        try {
            const response = await api.get<MenuItemResponse>(`/menu/menuItems/${id}`);
            return response.data;
        } catch (error: any) {
            if (error.response && error.response.status === 404) {
                return null;
            }
            
            console.error(`MenuService Error (id: ${id}):`, error);
            return null;
        }
    }
};

File: PolishPizza/menu/src/main/java/pizzeria/menu/menuItem/service/MenuItemService.java
package pizzeria.menu.menuItem.service;

import pizzeria.menu.menuItem.dto.request.MenuItemPatchRequest;
import pizzeria.menu.menuItem.dto.request.MenuItemRequest;
import pizzeria.menu.menuItem.dto.response.MenuItemResponse;

import java.util.List;
import java.util.UUID;

public interface MenuItemService {
    List<MenuItemResponse> getAllMenuItems();

    List<MenuItemResponse> getAvailableMenuItems();

    MenuItemResponse getMenuItemById(UUID id);

    MenuItemResponse save(MenuItemRequest request);

    void delete(UUID id);

    MenuItemResponse update(UUID id, MenuItemRequest request);

    MenuItemResponse patch(UUID id, MenuItemPatchRequest request);
}
